raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -145,13 +154,15 @@
     public KafkaRaftClient(RaftConfig raftConfig,
                            NetworkChannel channel,
                            ReplicatedLog log,
+                           TopicPartition metadataTopicPartition,
                            QuorumState quorum,
                            Time time,
                            FuturePurgatory<Void> purgatory,
                            InetSocketAddress advertisedListener,
                            LogContext logContext) {
         this(channel,
             log,
+            metadataTopicPartition,
```

Comment:
Could we encapsulate the TopicPartition information inside `ReplicatedLog` instead of in the RaftClient, and then when handling describe-quorum access the log for this information? I feel it is a bit cleaner since we only need this piece for that request handling only.

Reply:
My thought on this matter is that it's probably better to push partition information to lower level, as we haven't decided what Multi-raft looks like yet. Either 1:1 mapping from RaftClient to log, or 1:many mapping, it would make sense in both scenarios.